cairo/src/order_manager.cairo

```diff
@@ -129,101 +148,135 @@ mod OrderManagerComponent {
             };
 
             let positions = IPositionsDispatcher {
-                contract_address: contract_address_const::<0x06a2aee84bb0ed5dded4384ddd0e40e9c1372b818668375ab8e3ec08807417e5>()
+                contract_address: contract_address_const::<
+                    0x06a2aee84bb0ed5dded4384ddd0e40e9c1372b818668375ab8e3ec08807417e5,
+                >(),
             };
 
             let (id, minted) = positions.mint_and_increase_sell_amount(order_key, amount_u128);
-            
-            assert(minted != 0, ERROR_NO_TOKENS_MINTED);
-            assert(id != 0, ERROR_ZERO_AMOUNT);
 
             let sender: EthAddress = message.sender.try_into().unwrap();
-            
+
             // Store order data
-            self.order_id_to_depositor.write(id, sender);
-            self.order_depositor_to_id.write(sender, id);
-            self.order_depositor_to_order_created.write(
-                sender, 
-                Order_Created { order_key: order_key_copy, id }
-            );
+            self.order_id_to_creator.write(id, sender);
+            self
+                .order_id_to_order_created
+                .write(id, Order_Created { order_key: order_key_copy, creator: sender });
```

Comment:
this is redudant no? If im writing order id to creator then Im saving and id to the sender address

However how is this different from the second mapping which also has the order_Id and the creators address as well

Cant we just use the second mapping and when we check the order_ID for withdrawals we verify Order_Created.creator == depositor??